### ISSUE TITLE
Error: ETXTBSY: text file is busy

### ISSUE THREAD

**Issue by tugberkugurlu** [OmniSharp/omnisharp-vscode#565]
## Environment data

`dotnet --info` output:

```
.NET Command Line Tools (1.0.0-preview2-003121)

Product Information:
 Version:            1.0.0-preview2-003121
 Commit SHA-1 hash:  1e9d529bc5

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 RID:         ubuntu.14.04-x64

```

VS Code version:

```
Version 1.3.0
Commit e724f269ded347b49fcf1657fc576399354e6703
Date 2016-07-07T16:50:12.291Z
Shell 0.37.6
Renderer 49.0.2623.75
Node 5.10.0
```

C# Extension version: 1.2.2
## Steps to reproduce

not sure. I switched project and it happened. I managed to capture the log. It started working when I restarted Omnisharp.
## Expected  behavior

not have any errors
## Actual behavior

```
[INFO] Starting OmniSharp at '/home/tugberk/apps/foo/utils/rrcs'...
[INFO] Installing to /home/tugberk/.vscode/extensions/ms-vscode.csharp-1.2.2/.omnisharp
[INFO] Attempting to download omnisharp-1.9-beta11-ubuntu14-x64-netcoreapp1.0.tar.gz...
[INFO] Downloading to /tmp/tmp-320739ZvzKkiljjt.tmp...
[INFO] Starting OmniSharp at '/home/tugberk/apps/foo/src/foo-http-frontend'...
[INFO] Installing to /home/tugberk/.vscode/extensions/ms-vscode.csharp-1.2.2/.omnisharp
[INFO] Attempting to download omnisharp-1.9-beta11-ubuntu14-x64-netcoreapp1.0.tar.gz...
[INFO] Downloading to /tmp/tmp-3207f1smYjeIBhHY.tmp...
[INFO] Download complete!
[INFO] Decompressing...
[INFO] Download complete!
[INFO] Decompressing...
[INFO] Done! 323 files unpacked.
[INFO] Started OmniSharp from '/home/tugberk/.vscode/extensions/ms-vscode.csharp-1.2.2/.omnisharp/OmniSharp' with process id 3575...
OmniSharp: -s /home/tugberk/apps/foo/src/foo-http-frontend --hostPID 3207 dnx:enablePackageRestore=false --stdio
[INFORMATION:OmniSharp.Startup] Omnisharp server running using Stdio at location '/home/tugberk/apps/foo/src/foo-http-frontend' on host 3207.
[INFORMATION:OmniSharp.DotNet.DotNetProjectSystem] Initializing in /home/tugberk/apps/foo/src/foo-http-frontend
[INFORMATION:OmniSharp.DotNet.DotNetProjectSystem] Auto package restore: True
[INFORMATION:OmniSharp.DotNet.DotNetProjectSystem] Update workspace context
[ERROR] Error: ETXTBSY: text file is busy, open '/home/tugberk/.vscode/extensions/ms-vscode.csharp-1.2.2/.omnisharp/OmniSharp'
[ERROR] Error: ETXTBSY: text file is busy, open '/home/tugberk/.vscode/extensions/ms-vscode.csharp-1.2.2/.omnisharp/OmniSharp'
```

**Comment by DustinCampbell** [comment]
This really weird. I'm not sure what would cause that problem. Are you running in a VM, or is there anything else interesting about your setup?

**Comment by tugberkugurlu** [comment]
It's on a VM, I have only seen it once.

**Comment by DustinCampbell** [comment]
Thanks! I was just curious because I found several references to running on VMs where searching for this particular error.

It looks to me like you opening a project and it attempted to acquire OmniSharp (which it only does once). Then, you switched the project before it was finished, and it started trying to acquire it again without cancelling the previous attempt. So, they ran into conflict when the second download tried to overwrite the first one when it was already running. Thanks very much for the report! This is a good bug.

**Comment by DustinCampbell** [comment]
This should be fixed with 1.5 release since we'll acquire OmniSharp before attempting to launch it.

**Comment by tugberkugurlu** [comment]
:tada:
